src/profile/helpers/__tests__/updateMutationVariables.test.ts

```diff
@@ -106,112 +108,251 @@ const newAddress = {
+      getPhoneAsComparisonObject(updatedSecondaryPhone),
+    ]);
+  });
+
+  test('update arrays are empty when using unchanged or new data', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      {
+        ...formValues,
+        addresses: [primaryAddress, newAddress],
+        emails: [primaryEmail, newEmail],
+        phones: [primaryPhone, newPhone],
+      },
+      myProfile
+    );
+
+    expect(variables.input.profile.updateAddresses).toEqual([]);
+    expect(variables.input.profile.updateEmails).toEqual([]);
+    expect(variables.input.profile.updatePhones).toEqual([]);
+  });
+
+  test('remove arrays are formed correctly with existing data', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      { ...formValues, addresses: [], emails: [], phones: [] },
+      myProfile
+    );
+    expect(variables.input.profile.removeAddresses).toEqual([
+      primaryAddress.id,
+      updatedSecondaryAddress.id,
+    ]);
+    expect(variables.input.profile.removeEmails).toEqual([
+      primaryEmail.id,
+      updatedSecondaryEmail.id,
+    ]);
+    expect(variables.input.profile.removePhones).toEqual([
+      primaryPhone.id,
+      updatedSecondaryPhone.id,
+    ]);
+  });
+
+  test('remove arrays do not exists when data is not changed', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      formValues,
+      myProfile
+    );
+
+    expect(variables.input.profile.removeAddresses).toBeFalsy();
+    expect(variables.input.profile.removeEmails).toBeFalsy();
+    expect(variables.input.profile.removePhones).toBeFalsy();
+  });
 });
 
-test('remove arrays do not exists when data is not changed', () => {
-  const variables: UpdateMyProfileVariables = updateMutationVariables(
-    formValues,
-    myProfile
-  );
+describe('Full profile can be updated ', () => {
+  let formValues: Partial<FormValues>;
+
+  beforeEach(() => {
+    formValues = getFormValues([
+      ...multiNodeFormParts,
+      ...nameFormParts,
+      'language',
+    ]);
+  });
+
+  test('Variables include all formValues or corresponding update property', () => {
+    const variables: UpdateMyProfileVariables = updateMutationVariables(
+      formValues,
+      myProfile
+    );
+    expect(variables.input.profile.firstName).toBeDefined();
+    expect(variables.input.profile.nickname).toBeDefined();
+    expect(variables.input.profile.lastName).toBeDefined();
+    expect(variables.input.profile.language).toBeDefined();
+    expect(variables.input.profile.updateAddresses).toBeDefined();
+    expect(variables.input.profile.updateEmails).toBeDefined();
+    expect(variables.input.profile.updatePhones).toBeDefined();
+  });
 
-  expect(variables.input.profile.removeAddresses).toBeFalsy();
-  expect(variables.input.profile.removeEmails).toBeFalsy();
-  expect(variables.input.profile.removePhones).toBeFalsy();
+  test('but empty update is also handled', () => {
```

Comment:
This feels like a side sentence to another sentence? It might be good idea to make each test and its name independent of other test names for clarity. If the other test is then removed or modified, this one might not make as much sense anymore. Maybe just remove the first word _empty_?

Reply:
Fixed.